esmvaltool/diag_scripts/shared/_base.py

```diff
@@ -443,28 +441,38 @@ def main(cfg):
     logger.info("Starting diagnostic script %s with configuration:\n%s",
                 cfg['script'], yaml.safe_dump(cfg))
 
-    # Create output directories
+    # Clean run_dir and output directories from previous runs
+    default_files = ['log.txt', 'resource_usage.txt', 'settings.yml']
+
     output_directories = []
     if cfg['write_netcdf']:
         output_directories.append(cfg['work_dir'])
     if cfg['write_plots']:
         output_directories.append(cfg['plot_dir'])
 
-    existing = [p for p in output_directories if os.path.exists(p)]
-
-    if existing:
-        if args.force:
-            for output_directory in existing:
-                logger.info("Removing %s", output_directory)
-                shutil.rmtree(output_directory)
-        elif not args.ignore_existing:
-            logger.error(
-                "Script will abort to prevent accidentally overwriting your "
-                "data in these directories:\n%s\n"
-                "Use -f or --force to force emptying the output directories "
-                "or use -i or --ignore-existing to ignore existing output "
-                "directories.", '\n'.join(existing))
+    old_content = [p for p in output_directories if os.path.exists(p)]
+    old_content += [
+        p for p in glob.glob(f"{cfg['run_dir']}/*")
+        if not os.path.basename(p) in default_files
+    ]
+
+    if old_content:
+        for content in old_content:
+            if args.force:
+                logger.info("Removing %s", content)
+                if os.path.isfile(content):
+                    os.remove(content)
+                else:
+                    shutil.rmtree(content)
+            elif not args.ignore_existing:
+                raise FileExistsError(
+                    "Script will abort to prevent accidentally overwriting your "
+                    "data in the following output files or directories:\n%s\n"
+                    "Use -f or --force to force emptying the output directories "
+                    "or use -i or --ignore-existing to ignore existing output "
+                    "directories." % '\n'.join(old_content))
```

Comment:
```suggestion
        if args.force:
            for content in old_content:
                logger.info("Removing %s", content)
                if os.path.isfile(content):
                    os.remove(content)
                else:
                    shutil.rmtree(content)
        elif not args.ignore_existing:
            raise FileExistsError(
                "Script will abort to prevent accidentally overwriting your "
                "data in the following output files or directories:\n%s\n"
                "Use -f or --force to force emptying the output directories "
                "or use -i or --ignore-existing to ignore existing output "
                "directories." % '\n'.join(old_content))
```

Reply:
Why switch the order of the for loop and the if?

Reply:
No reason. I first tried a different approach and then didn't change it back. Now I did.